src/table/system.rs

```diff
@@ -1,19 +1,24 @@
+use core::marker::PhantomData;
+use core::slice;
+
+use crate::proto::console::text;
+use crate::{CStr16, Char16, Handle, Result, ResultExt, Status};
+
 use super::boot::{BootServices, MemoryMapIter};
 use super::runtime::RuntimeServices;
 use super::{cfg, Header, Revision};
-use crate::proto::console::text;
-use crate::{CStr16, Char16, Handle, Result, ResultExt, Status};
-use core::marker::PhantomData;
-use core::slice;
 
 /// Marker trait used to provide different views of the UEFI System Table
 pub trait SystemTableView {}
 
 /// Marker struct associated with the boot view of the UEFI System Table
+#[derive(Clone)]
```

Comment:
Any reason for making these cloneable?

Reply:
yes I used clone to allow moving of system table by cloning it, maybe I should change it to reference somehow🤔

Reply:
Note that this breaks the API design used to make the SystemTable API safe...

Right now, the API encodes in the type system that once you transition away from boot services, you don't have access to them anymore. If you are able to clone the boot services before performing the transition, then you still have access to them, and using this access will trigger undefined behavior.